python/cudf/cudf/core/column/numerical.py

```diff
@@ -109,13 +99,28 @@ def binary_operator(
                 msg = "{!r} operator not supported between {} and {}"
                 raise TypeError(msg.format(binop, type(self), type(rhs)))
             out_dtype = np.result_type(self.dtype, rhs.dtype)
-            if binop in ["mod", "floordiv"]:
-                tmp = self if reflect else rhs
-                if (tmp.dtype in int_dtypes) and (
-                    (np.isscalar(tmp) and (0 == tmp))
-                    or ((isinstance(tmp, NumericalColumn)) and (0.0 in tmp))
-                ):
-                    out_dtype = np.dtype("float64")
+
+            # Special cases for mod and floordiv
+            if binop in {"mod", "floordiv"}:
+                denominator = self if reflect else rhs  # denominator
+                numerator = rhs if reflect else self  # numerator
+                # check for zeroes in the denominator
+                if denominator.dtype.kind in "bui":
+                    # Denominator either zero, or zero is contained therein
+                    if (np.isscalar(denominator) and denominator == 0) or (
```

Comment:
Shouldn't use `np.isscalar` here because we could have a `cudf.Scalar` here, no?